src/Context/GtmContext.php

```diff
@@ -75,17 +76,58 @@ public function getDataLayerSettingShouldMatch($key, $regex) {
    * @return mixed
    * @throws \Exception
    */
-  protected function getDataLayerValue($key) {
-    $json_arr = $this->getDataLayerJson();
+    protected function getDataLayerValue($key)
+    {
+        $json_arr = $this->getDataLayerJson();
+        // Loop through the array and return the data layer value
+        foreach ($json_arr as $json_item) {
+            // Check if the key contains dot.
+            if (strpos($key, '.', 0) !== false) {
+                $value = $this->getDotValue($json_item, $key);
+                if (!is_null($value)) {
+                    return $value;
+                }
+            } elseif (isset($json_item[$key])) {
+                return $json_item[$key];
+            }
+        }
+        throw new \Exception($key . ' not found.');
+    }
 
-    // Loop through the array and return the data layer value
-    foreach ($json_arr as $json_item) {
-      if (isset($json_item[$key])) {
-        return $json_item[$key];
-      }
+    /**
+     * Convert dataLayer values to dot notation
+     */
+    private function getDotValue($arr, $key)
+    {
+        $dot = dot($arr);
+        // Check if the array contains dot.
```

Comment:
This comment is not accurate. It should say `// Check if the key contains [.
We also need to add a comment in the main Readme to explain how to use this into a Feature file.

Reply:
Please update Readme with this new feature https://github.com/marcelovani/behat-gtm/blob/54c1b9f5ebd29c946afb1e1c8722ed7aae90ec33/README.md